### DISCUSSION TITLE
Parameters for deconvolution

### DISCUSSION THREAD

**Discussion by AlexanderMoysa** [michaelmarty/UniDec#103] [Q&A] [answered]
Hi Michael, I have a question. I would like to process data using Unidec/Pycharm.
import unidec  
file_name="test.txt"
folder="C:\\data"  
eng=unidec.UniDec() 
eng.open_file(file_name, folder)  
eng.process_data() eng.run_unidec(silent=True) 
eng.pick_peaks()
However, I cannot understand how to set the parameters of deconvolution.  Before I used GUI Windows version. Can I import parameters from conf file to python? I cannot find this information in the documentation.
Thank you for your support!
Best,
Alex

**Comment by michaelmarty** [comment, marked as answer]
Hi Alex,
Great question. eng.load_config(<filename>) will load a config file. You can also do eng.config.<parameter name> = <value> to adjust specific parameters. unidecstructure.py has all the parameters listed. There's a lot, but hopefully you can track them down. Happy to help if there are questions on any of them. Thanks,
MTM

**Reply by AlexanderMoysa** [reply]
Hi Michael! Thank you very much!

**Reply by michaelmarty** [reply]
Looks like some of my text markings didn't work in mark down but you figured it out. For clarity on anyone in the future looking at this, I meant to use eng.config.load_config(filename) and eng.config.parametername=value. Will avoid >< brackets next time :)

**Reply by AlexanderMoysa** [reply]
Hi Michael! Thank you!

**Comment by AlexanderMoysa** [comment]
I wrote this code, and it works successfully. It utilizes the conf.dat file from the UniDec GUI (found in the folder containing the data analysis results). So, it is possible to adjust parameters using UniDec GUI and use it for the processing large sets of data using Python. Perhaps this might be useful to someone. 

```
# Import libraries
import unidec
import os
import logging

# Set up logging
logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")

# The folder path and configuration file path
folder = r"C:\data"
config_file = os.path.join(folder, "conf.dat")

# Initialize the UniDec engine
eng = unidec.UniDec()

# Load configuration from conf.dat
try:
    eng.load_config(config_file)
    logging.info(f"Configuration loaded from {config_file}")
except Exception as e:
    logging.error(f"Failed to load configuration from {config_file}: {e}")
    raise

# Process each file in the folder
for file_name in os.listdir(folder):
    file_path = os.path.join(folder, file_name)  # Get full path to the file

    # Check if the file is a text file (or other specific extension, if needed)
    if file_name.endswith(".txt"):  # Adjust the condition for other file types if necessary
        logging.info(f"Processing file: {file_name}")

        try:
            eng.open_file(file_path)  # Open the current file

            # Process the data, run UniDec, and pick peaks
            eng.process_data()
            eng.run_unidec(silent=True)
            eng.pick_peaks()
            logging.info(f"Successfully processed: {file_name}")

        except Exception as e:
            logging.error(f"Error processing file {file_name}: {e}")
```
